### DISCUSSION TITLE
Extracting/injecting rpu clarification

### DISCUSSION THREAD

**Discussion by Travillion** [quietvoid/dovi_tool#78] [Q&A]
I apologize in advance for the noob question. I'm still very new to Dolby Vision. I have a single track dual layer (profile 7) MKV straight out of makemkv. In order to reencode with x265 and keep DV, I would need to extract the DV metadata from the original and inject it into the reencoded file. I believe I can do this using the extract and inject commands. I'm not trying to convert profiles or anything.

In doing this, am I reconstructing the DV perfectly? I ask because the documentation I've looked at suggests the extract/inject commands only involve the RPU data, and I'm not sure how that relates to the BL, MEL, and FEL. Are there additional commands I need to use?

Thanks in advance.

**Comment by quietvoid** [comment]
x265 only supports profile 8.1 for reencoding an HDR10 base layer.
So you should convert the RPU.

The FEL info is indeed lost that way.

**Comment by Travillion** [comment]
If HDR10 isn't retained, and thus not converted to profile 8.1, will
extract/inject maintain FEL?

**Reply by quietvoid** [reply]
I don't understand.
Profile 7 is a HDR10 base layer, it is always present.

**Comment by Travillion** [comment]
I'm sure it's me who doesn't understand!

I've been backing up my BR movies for quite some time but I didn't have a
4k tv until now and so I'm just getting started working with uhd discs and
HDR formats.

So the base layer (BL) is the HDR10 metadata? And the DV metadata is the EL
(MEL and sometimes FEL)? When extracting the RPU with dovi_tool, is it also
grabbing the BL, or is it only grabbing the MEL?

**Reply by quietvoid** [reply]
Extracting the RPU does just that, extracts the RPU.
MEL/FEL are composed of another video stream AND the RPU.

So if you encode with the RPU only, you end up with a (reencoded) HDR10 base layer, with the Dolby Vision metadata from the RPU.
No more MEL/FEL.

**Comment by Travillion** [comment]
I see. Thank you for all your hard work on developing these tools!!
